File: examples/synthetic-java-service/src/main/java/com/example/mercurio/customer/CustomerKafkaReferences.java
package com.example.mercurio.customer;

import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
import org.apache.kafka.clients.admin.NewTopic;
import org.springframework.kafka.core.KafkaTemplate;

public final class CustomerKafkaReferences {
    private final KafkaTemplate<String, String> kafkaTemplate;
    private final SchemaRegistryClient schemaRegistryClient;

    public CustomerKafkaReferences(
            KafkaTemplate<String, String> template,
            SchemaRegistryClient registry) {
        this.kafkaTemplate = template;
        this.schemaRegistryClient = registry;
    }

    public NewTopic customerUpdatedTopic() {
        return new NewTopic("customer.updated", 1, (short) 1);
    }

    public void publish(String payload) {
        kafkaTemplate.send("customer.updated", payload);
    }

    public void inspectSchema() {
        schemaRegistryClient.getLatestSchemaMetadata("customer-updated-value");
    }

    public String unrelatedTopicLikeString() {
        return "customer.unrelated";
    }
}

File: examples/synthetic-kotlin-service/src/main/kotlin/com/example/mercurio/customer/CustomerKafkaReferences.kt
package com.example.mercurio.customer

import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient
import org.apache.kafka.clients.admin.NewTopic
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.kafka.core.KafkaTemplate

class CustomerKafkaReferences(
    private val kafkaTemplate: KafkaTemplate<String, String>,
    private val schemaRegistryClient: SchemaRegistryClient,
) {
    fun customerUpdatedTopic(): NewTopic = NewTopic("customer.updated", 1, 1.toShort())

    fun publish(payload: String) {
        kafkaTemplate.send("customer.updated", payload)
    }

    @KafkaListener(topics = ["customer.updated"])
    fun receive(payload: String) = payload

    fun inspectSchema() {
        schemaRegistryClient.getLatestSchemaMetadata("customer-updated-value")
    }

    fun unrelatedTopicLikeString(): String = "customer.unrelated"
}
